src/main/rx/srxl2.c

```diff
@@ -0,0 +1,558 @@
+static uint32_t lastValidPacketTimestamp = 0;
+static volatile uint32_t lastReceiveTimestamp = 0;
+static volatile uint32_t lastIdleTimestamp = 0;
+
+static Srxl2Frame readBuffer[2];
+static Srxl2Frame* readBufferPtr = &readBuffer[0];
+static Srxl2Frame* processBufferPtr = &readBuffer[1];
+static volatile uint8_t readBufferIdx = 0;
+static volatile uint8_t readBytes = 0;
+static volatile bool transmittingTelemetry = false;
+static uint8_t writeBuffer[SRXL2_MAX_PACKET_LENGTH];
+static uint8_t writeBufferIdx = 0;
+
+static serialPort_t *serialPort;
+
+static uint8_t busMasterDeviceId = 0xFF;
+static bool telemetryRequested = false;
+
+static uint8_t telemetryFrame[22];
+
+uint8_t globalResult = 0;
+
+/* handshake protocol
+    1. listen for 50ms for serial activity and go to State::Running if found, autobaud may be necessary
+    2. if srxl2_unitId = 0:
+            send a Handshake with destinationDeviceId = 0 every 50ms for at least 200ms
+        else:
+            listen for Handshake for at least 200ms
+    3.  respond to Handshake as currently implemented in process if rePst received
+    4.  respond to broadcast Handshake
+*/
+
+// if 50ms with not activity, go to default baudrate and to step 1
+
+bool srxl2ProcessHandshake(const Srxl2Header* header)
+{
+    const Srxl2HandshakeSubHeader* handshake = (Srxl2HandshakeSubHeader*)(header+1);
+    if (handshake->destinationDeviceId == Broadcast) {
+        DEBUG("broadcast handshake from %x\r\n", handshake->sourceDeviceId);
+        busMasterDeviceId = handshake->sourceDeviceId;
+
+        if (handshake->baudSupported == 1) {
+            serialSetBaudRate(serialPort, SRXL2_PORT_BAUDRATE_HIGH);
+            DEBUG("switching to %d baud\r\n", SRXL2_PORT_BAUDRATE_HIGH);
+        }
+
+        state = Running;
+
+        return true;
+    }
+
+
+    if (handshake->destinationDeviceId != ((FlightController << 4) | unitId)) {
+        return true;
+    }
+
+    DEBUG("FC handshake from %x\r\n", handshake->sourceDeviceId);
+
+    Srxl2HandshakeFrame response = {
+        .header = *header,
+        .payload = {
+            handshake->destinationDeviceId,
+            handshake->sourceDeviceId,
+            /* priority */ 10,
+            /* baudSupported*/ baudRate,
+            /* info */ 0,
+            U_ID_2
+        }
+    };
+
+    const uint16_t crc = crc16_ccitt_update(0, &response, sizeof(response) - 2);
+    response.crcHigh = (crc >> 8);
+    response.crcLow = (crc & 0xFF);
+
+    srxl2RxWriteData(&response, sizeof(response));
+
+    return true;
+}
+
+void srxl2ProcessChannelData(const Srxl2ChannelDataHeader* channelData, rxRuntimeConfig_t *rxRuntimeConfig) {
+    if (channelData->rssi >= 0) {
+        const int rssiPercent = channelData->rssi;
+        setRssi(scaleRange(rssiPercent, 0, 100, 0, RSSI_MAX_VALUE), RSSI_SOURCE_RX_PROTOCOL);
+    } else {
+        // If dBm value provided, cant properly convert to % without knowing the receivers sensitivity range. Fix at 50% for now.
```

Comment:
My concern here is that having a built in conversion, even if not the same as our provided conversion, might not be enough to make users feel like something is wrong. If they have the option to change the receivers output to dBm instead of %, then they should be able to set it back. By default, all new receivers should be providing %, and we may give them the option to change this somehow down the line. If rssi doesn't work at all, that is a clear indicator to them that something is not set up correctly, and something has to be changed. I prefer this over inconsistent behavior without them fully realizing it. It would be nicer if betaflight could show dBm instead of % is that is what the receiver is providing.